pkg/cloudclient/aws/aws.go

```diff
@@ -53,6 +53,18 @@ func (c *Client) VerifyDns(ctx context.Context, vpcID string) *output.Output {
 // NewClient creates a new CloudClient for use with AWS.
 func NewClient(ctx context.Context, logger ocmlog.Logger, creds interface{}, region, instanceType string, tags map[string]string) (client *Client, err error) {
 	switch c := creds.(type) {
+	case string:
+		client, err = newClient(
+			ctx,
+			logger,
+			"",
+			"",
+			"",
+			region,
+			instanceType,
+			tags,
+			fmt.Sprintf("%v", creds),
```

Comment:
this can be `creds.(string),` as we're sure that the type of it is a `string`. I think there's no need to use fmt